### PR TITLE
Make ChannelActionListener a top-level class

### PR DESCRIPTION
**Pull request by dnhatn**
We start using this class more often. Let's make it a top-level class.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/action/support/ChannelActionListener.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+import org.apache.logging.log4j.message.ParameterizedMessage;
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.transport.TransportChannel;
+import org.elasticsearch.transport.TransportRequest;
+import org.elasticsearch.transport.TransportResponse;
+
+public final class ChannelActionListener<
+    Response extends TransportResponse, Request extends TransportRequest> implements ActionListener<Response> {
+
+    private final Logger logger = LogManager.getLogger(getClass());
```
Comment (original-brownbear) [inline server/src/main/java/org/elasticsearch/action/support/ChannelActionListener.java]: I know it was like this before, but maybe make this `logger` `static`?

**Review by original-brownbear — APPROVED** [review]
LGTM, one suggested cleanup (since you'll have to merge in `master` anyway to make it green :)).

File: server/src/main/java/org/elasticsearch/action/support/ChannelActionListener.java (hunk 1 above)
Comment (dnhatn) [reply server/src/main/java/org/elasticsearch/action/support/ChannelActionListener.java]: I pushed https://github.com/elastic/elasticsearch/pull/37797/commits/a0afac5c9586d02ada5f1e515f899871acdfdb05.

**Comment by dnhatn** [comment]
Thanks @original-brownbear.

**Pull request merged**